src/identifiers.md

```diff
@@ -29,7 +29,7 @@ The HIR uses a bunch of different identifiers that coexist and serve different p
 - A [`DefId`], as the name suggests, identifies a particular definition, or top-level
   item, in a given crate. It is composed of two parts: a [`CrateNum`] which identifies
   the crate the definition comes from, and a [`DefIndex`] which identifies the definition
-  within the crate. Unlike [`NodeId`]s, there isn't a [`DefId`] for every expression, which
+  within the crate. Unlike [`HirId`]s, there isn't a [`DefId`] for every expression, which
```

Comment:
At this point the guide hasn't discussed HirIds yet. Do you want to move that section above DefIds? It's just a couple bullet points below.

Reply:
There's an infinite cycle: the `DefId` bullet refer to `HirId` and the `HirId` bullet refers to `DefId` :)

Also, `HirId` refers (indirectly) to `LocalDefId`, which is above it and below `DefId`. I'm fine to move it though; I don't think it matters too much either way.

Reply:
Fair enough, this seems fine then.